src/Octoshift/Services/GithubApi.cs

```diff
@@ -70,10 +70,30 @@ public virtual async Task DeleteAutoLink(string org, string repo, int autoLinkId
         var url = $"{_apiUrl}/orgs/{org.EscapeDataString()}/teams";
         var payload = new { name = teamName, privacy = "closed" };
 
-        var response = await _client.PostAsync(url, payload);
-        var data = JObject.Parse(response);
-
-        return ((string)data["id"], (string)data["slug"]);
+        return await _retryPolicy.HttpRetry(async () =>
+        {
+            try
+            {
+                var response = await _client.PostAsync(url, payload);
+                var data = JObject.Parse(response);
+                return ((string)data["id"], (string)data["slug"]);
+            }
+            catch (HttpRequestException ex) when (ex.StatusCode >= HttpStatusCode.InternalServerError)
+            {
+                // Before retrying, check if the team was actually created
+                var teams = await GetTeams(org);
+                var (Name, Slug) = teams.FirstOrDefault(t => t.Name == teamName);
+                if (Name != null)
+                {
+                    // Team exists, return its details instead of retrying
+                    var teamResponse = await _client.GetAsync($"{_apiUrl}/orgs/{org.EscapeDataString()}/teams/{Slug.EscapeDataString()}");
```

Comment:
Instead of making another API call, let's improve the GetTeams function to include ID in it's return value.